routing/localchans/manager_test.go

```diff
@@ -115,34 +138,173 @@ func TestManager(t *testing.T) {
+					},
+					active: true,
+				},
+				{
+					edgeInfo: &channeldb.ChannelEdgeInfo{
+						Capacity:     chanCap,
+						ChannelPoint: chanPointInactive,
+					},
+					active: false,
+				},
+			},
+			specifiedChanPoints: []wire.OutPoint{
+				chanPointValid,
+				chanPointInactive,
+			},
+			expectedNumUpdates: 1,
+			expectedUpdateFailures: []lnrpc.UpdateFailure{
+				lnrpc.UpdateFailure_UPDATE_FAILURE_INACTIVE,
+			},
+		},
+		{
+			name:          "all channels",
+			currentPolicy: currentPolicy,
+			newPolicy:     newPolicy,
+			channelSet: []channel{
+				{
+					edgeInfo: &channeldb.ChannelEdgeInfo{
+						Capacity:     chanCap,
+						ChannelPoint: chanPointValid,
+					},
+					active: true,
+				},
+			},
+			specifiedChanPoints:    []wire.OutPoint{},
+			expectedNumUpdates:     1,
+			expectedUpdateFailures: []lnrpc.UpdateFailure{},
+		},
+		{
+			name:          "missing channel",
+			currentPolicy: currentPolicy,
+			newPolicy:     newPolicy,
+			channelSet: []channel{
+				{
+					edgeInfo: &channeldb.ChannelEdgeInfo{
+						Capacity:     chanCap,
+						ChannelPoint: chanPointValid,
+					},
+					active: true,
+				},
+			},
+			specifiedChanPoints: []wire.OutPoint{chanPointMissing},
+			expectedNumUpdates:  0,
+			expectedUpdateFailures: []lnrpc.UpdateFailure{
+				lnrpc.UpdateFailure_UPDATE_FAILURE_NOTFOUND,
+			},
+		},
+		{
+			// Here, no max htlc is specified, the max htlc value
+			// should be kept unchanged.
+			name:          "no max htlc specified",
+			currentPolicy: MaxHTLCPolicy,
+			newPolicy:     noMaxHtlcPolicy,
+			channelSet: []channel{
+				{
+					edgeInfo: &channeldb.ChannelEdgeInfo{
+						Capacity:     chanCap,
+						ChannelPoint: chanPointValid,
+					},
+					active: true,
+				},
+			},
+			specifiedChanPoints:    []wire.OutPoint{chanPointValid},
+			expectedNumUpdates:     1,
+			expectedUpdateFailures: []lnrpc.UpdateFailure{},
+		},
 	}
 
-	// If no max htlc is specified, the max htlc value should be kept
-	// unchanged.
-	currentPolicy.MaxHTLC = newPolicy.MaxHTLC
-	noMaxHtlcPolicy := newPolicy
-	noMaxHtlcPolicy.MaxHTLC = 0
+	for _, test := range tests {
+		test := test
+		t.Run(test.name, func(t *testing.T) {
+			currentPolicy = test.currentPolicy
+			channelSet = test.channelSet
+			expectedNumUpdates = test.expectedNumUpdates
+
+			failedUpdates, err := manager.UpdatePolicy(test.newPolicy,
+				test.specifiedChanPoints...)
+
+			if len(failedUpdates) != len(test.expectedUpdateFailures) {
+				t.Fatalf("wrong number of failed policy updates")
+			}
+
+			if len(test.expectedUpdateFailures) > 0 {
+				if failedUpdates[0].Reason != test.expectedUpdateFailures[0] {
+					t.Fatalf("wrong expected policy update failure")
+				}
+			}
 
-	err = manager.UpdatePolicy(noMaxHtlcPolicy)
-	if err != nil {
-		t.Fatal(err)
+			if err != nil && test.expectErr {
+				return
+			}
+			if err != nil {
+				t.Fatalf(err.Error())
+			}
+			if test.expectErr {
+				t.Fatalf("expected error")
+			}
```

Comment:
Could flatten all this into having an expected error `expectedErr error` in each case and just `require.Equal(t, test.ExpectedErr, err)`.

Reply:
Awesome.